zombies.js

```diff
@@ -79,6 +108,116 @@
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
 
+class Player {
+
+  constructor(name, health, strength, speed) {
+    typeErrors(name, 'string');
+    typeErrors(health, 'number');
+    typeErrors(strength, 'number');
+    typeErrors(speed, 'number');
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this._pack = [];
+    this._maxHealth = health;
+    this.isAlive = true;
+    this.equipped = false;
+    this.getPack = function () { return this._pack; };
+    this.getMaxHealth = function () { return this._maxHealth; };
+
+  }
+
+  checkPack() {
+    let playerPack = this.getPack();
+
+    if (playerPack.length > 0) {
+      for (let i = 0; i < playerPack.length; i++) {
+        console.log('Item' + (i + 1) + ': ' + playerPack[i].name);
+      }
+    } else {
+      console.log('The pack is empty.');
+    }
+  }
+
+  takeItem(item) {
+    let playerPack = this.getPack();
+
+    if (playerPack.length < 3) {
+      playerPack.push(item);
+      console.log(item.name + " was added to " + this.name + "'s pack.");
+      return true;
+    } else {
+      console.log('The pack is full. The item could not be stored.');
+      return false;
+    }
+  }
+
+  discardItem(item) {
+    let playerPack = this.getPack();
+    let indexOfRemoval = playerPack.indexOf(item);
+
+    if (indexOfRemoval > -1) {
+      playerPack.splice(indexOfRemoval, 1);
+      console.log(this.name + ' discarded ' + item.name +
+        ' from their pack.');
+      return true;
+    } else {
+      console.log(this.name + ' didn\'t find ' + item.name +
+        ' in their pack. Nothing was discarded.');
+      return false;
+    }
+  }
+
+  equip(itemToEquip) {
+
+    if ((itemToEquip instanceof Weapon)) {
+      let wasItInPack = this.discardItem(itemToEquip);
+
+      if ((this.equipped !== false) && (wasItInPack)) {
+        this.takeItem(this.equipped);
+        this.equipped = itemToEquip;
+      } else if ((this.equipped === false) && (wasItInPack)) {
+        this.equipped = itemToEquip;
+      }
+    }
+
+    return false;
+  }
+
+  eat(itemToEat) {
+    if ((itemToEat instanceof Food)) {
+
+      let wasItInPack = this.discardItem(itemToEat);
+
+      if (wasItInPack) {
+
+        let maxHealth = this.getMaxHealth();
+        this.health += itemToEat.energy;
+
+        if (this.health > maxHealth) {
+          this.health = maxHealth;
+        }
+      }
+    }
+  }
+
+  useItem(item) {
+    if (item instanceof Weapon) {
+      this.equip(item);
+    } else if (item instanceof Food) {
+      this.eat(item);
+    }
```

Comment:
It might not be needed for this exercise, but in general you would want to be able to handle all cases. So what if the item is not Weapon or Food.